examples/s1_generic.py

```diff
@@ -0,0 +1,309 @@
+        main_family = None
+        gene_count = None
+
+    note = f'{note}{emoji} {verdict.capitalize()}'
+
+    if main_family:
+        note = f'{note} - {main_family}'
+        if gene_count:
+            note = f'{note} ({gene_count} shared code genes)'
+    note = f'{note}\n\nSize: {human_readable_size(metadata["size_in_bytes"])}\n'
+    if 'file_type' in metadata:
+        note = f'{note}File type: {metadata["file_type"]}\n'
+    if verdict == 'malicious':
+        iocs = len(analysis.iocs['files']) + len(analysis.iocs['network']) - 1
+
+        if iocs:
+            note = f'{note}IOCs: {iocs} IOCs\n'
+        dynamic_ttps = len(analysis.dynamic_ttps)
+
+        if dynamic_ttps:
+            note = f'{note}TTPs: {dynamic_ttps} techniques\n'
+
+    else:
+        note = f'{note}Verdict: {verdict}\nSub-verdict: {sub_verdict}\n'
+
+    note = (f'{note}\nFull report\n'
+            f'👉{result["analysis_url"]}')
+
+    return note
+
+
+def send_note(threat_id: str, analysis: Analysis):
+    note = get_note(analysis)
+    response = _s1_session.post('/web/api/v2.1/threats/notes',
+                                json={'data': {'text': note}, 'filter': {'ids': [threat_id]}})
+    assert_s1_response(response)
+
+
+def send_failure_note(note: str, threat_id: str):
+    response = _s1_session.post('/web/api/v2.1/threats/notes',
+                                json={'data': {'text': note}, 'filter': {'ids': [threat_id]}})
+    assert_s1_response(response)
+
+
+def analyze_threat(intezer_api_key: str, s1_api_key: str, s1_base_address: str, threat_id: str):
+    api.set_global_api(intezer_api_key)
+    init_s1_requests_session(s1_api_key, s1_base_address)
+    _logger.info(f'incoming threat: {threat_id}')
+    try:
+        threat = get_threat(threat_id)
+        if not filter_threat(threat):
+            _logger.info(f'threat {threat_id} is been filtered')
+            return
+
+        threat_info = threat['threatInfo']
+        file_hash = threat_info.get('sha256') or threat_info.get('sha1') or threat_info.get('md5')
+        analysis = None
+        if file_hash:
+            _logger.debug(f'trying to analyze by hash {file_hash}')
+            try:
+                analysis = Analysis(file_hash=file_hash)
+                analysis.send()
+            except errors.HashDoesNotExistError:
+                _logger.debug(f'hash {file_hash} not found on server, fetching the file from endpoint')
+                analysis = None
+
+        if not analysis:
+            _logger.debug(f'Starting to analyze file')
+            analysis = analyze_by_file(threat_id)
+            analysis.send(requester='s1')
+
+        analysis.wait_for_completion()
+        send_note(threat_id, analysis)
+    except Exception as ex:
+        send_failure_note(str(ex), threat_id)
+
+
+def parse_argparse_args():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('-i', '--Intezer', help='Pass intezer API key', dest='intezer_api_key', required=True)
```

Comment:
the longer command line argument should be informative, so it should be `--intezer-api-key`

Reply:
Done